### ISSUE TITLE
Conditionals not supported in element overrides

### ISSUE THREAD

**Issue by Cynical-Optimist** [buildstream-migration/bst-staging#658]
[See original issue on GitLab](https://gitlab.com/BuildStream/buildstream/-/issues/658)
In GitLab by [[Gitlab user @abderrahimk]](https://gitlab.com/abderrahimk) on Sep 17, 2018, 21:34

## Summary

Trying to use a conditional inside an element override, I get an error: Value of '(?)' is not of the expected type 'str'

## Steps to reproduce

Here is a small project.conf to reproduce

```
name: testcase

options:
  arch:
    type: arch
    description: architecture
    values: [i686, x86_64]

elements:
  autotools:
    variables:
      (?):
      - arch == 'i686':
          conf-global: --host=i686-unknown-linux-gnu
      - arch == 'x86_64':
          conf-global: --host=x86_64-unknown-linux-gnu
```

## What is the current bug behavior?

```
project.conf [line 13 column 6]: Value of '(?)' is not of the expected type 'str'
```

## What is the expected correct behavior?

It should work

## Possible fixes

[//]: # (If you can, link to the line of code that might be responsible for the problem)

## Other relevant information

[//]: # (Add the BuildStream version. Substitute below the "x" by "1", "2" or the right version. Add more than one version if necessary)

* BuildStream version affected: /milestone %BuildStream_v1.2

----

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Sep 18, 2018, 08:48

mentioned in commit ffa0bb36367fc5b0e105e9d19344358b7d5ef3e3

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Sep 18, 2018, 08:48

mentioned in commit 727f2faa307938f95928ee1e7e4ce961e760d92e

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Sep 18, 2018, 08:51

mentioned in commit 758dbece739606669fb5e1beb45a3e9c0ef7f10c

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Sep 18, 2018, 08:51

mentioned in commit 360614ef6dac729e202b8b8050e6a573faa6c712

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Sep 18, 2018, 09:14

closed via merge request !802

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Sep 18, 2018, 09:14

mentioned in commit f2ae46f8e59212f78336cf3e4adfe4ac40095110

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Sep 18, 2018, 09:56

Thanks for reporting this succinctly [[Gitlab user @abderrahimk]](https://gitlab.com/abderrahimk) !

I have fix this in !802 and !803 (for `bst-1.2`), and will be releasing a `1.2.1` this week including a few other important fixes.
